osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -264,20 +276,37 @@ public void SetHyperDashState(double modifier = 1, float targetPosition = -1)
 
                 if (!wasHyperDashing)
                 {
-                    this.FadeColour(Color4.OrangeRed, hyper_dash_transition_length, Easing.OutQuint);
-                    this.FadeTo(0.2f, hyper_dash_transition_length, Easing.OutQuint);
-                    Trail = true;
+                    updateCatcherColour(true);
 
-                    var hyperDashEndGlow = createAdditiveSprite();
-
-                    hyperDashEndGlow.MoveToOffset(new Vector2(0, -10), 1200, Easing.In);
-                    hyperDashEndGlow.ScaleTo(hyperDashEndGlow.Scale * 0.95f).ScaleTo(hyperDashEndGlow.Scale * 1.2f, 1200, Easing.In);
-                    hyperDashEndGlow.FadeOut(1200);
-                    hyperDashEndGlow.Expire(true);
+                    trails.DisplayTrail = true;
+                    trails.DisplayEndGlow();
                 }
             }
         }
 
+        private void updateCatcherColour(bool hyperDashing)
+        {
+            if (hyperDashing)
+            {
+                // special behaviour for catcher colour if no skin overrides.
+                var catcherColour =
+                    hyperDashColour == DEFAULT_HYPER_DASH_COLOUR
+                        ? DEFAULT_CATCHER_HYPER_DASH_COLOUR
+                        : hyperDashColour;
```

Comment:
I find this behaviour of silently overriding the colour from `Red` to `OrangeRed` really confusing. It complicates everything (including test code) and leads to unexpected behaviour (the user setting a red colour is now indistinguishable from the default - in both cases this colour will get silently converted to `OrangeRed`). I propose we just get rid of it for simplicity.

Reply:
I agree, I've tried comparing them with each other, the `Red` colour made the sprite more red-er than `OrangeRed`, but since the catcher is small I don't think it'll be that notable.

Reply:
i'm fine with just removing it if you can't figure out how to write tests. your call.

Reply:
The only 2 sane ways I can provide to implement this would be:
 1. Do some type of calculation to get a colour that can blend with the trails similar to how `Color4.Red` blends with `Color4.OrangeRed(alpha: 0.2f)`, sounds nearly impossible or just way too complicated to even be done correctly, *not even sure to call it a sane way*.
2. Rename `hyperDashColour` to `customHyperDashColour` and let it fallback to `DEFAULT_CATCHER_HYPER_DASH_COLOUR` on null here, that's a bit more readable and will solve the issue @bdach stated above, but the effect will only come here. as for the tests, I'll replay in the below conversation about it.

If nothing here sounds sane to you then I honestly don't know, may be that I can't really be more helpful in this specific case or that this is just way too complicated to be simplified.